library/Reporting/Model/Timeframe.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Icinga\Module\Reporting\Model;
+
+use ipl\Orm\Model;
+use ipl\Orm\Relations;
+
+class Timeframe extends Model
+{
+    public function getTableName()
+    {
+        return 'timeframe';
+    }
+
+    public function getKeyName()
+    {
+        return 'id';
+    }
+
+    public function getColumns()
+    {
+        return [
+            'name',
+            'title',
+            'start',
+            'end',
+            'ctime',
+            'mtime'
+        ];
+    }
+
+    public function createRelations(Relations $relations)
+    {
+        $relations->hasOne('report', Report::class);
```

Comment:
Are you sure about the relation type? Can't you create multiple reports with the same timeframe?